Allow the post type dropdown to work

hirozed opened the pull request:
![post-type-selector](https://github.com/user-attachments/assets/75d34b31-25c6-4786-a370-86b52c130443)

This updates the component to set an array of post types. The search term will carry over to each post type.

hirozed committed:
Update Post Chooser to allow a selection of Post Types

hirozed committed:
One small tweak

acketon commented on components/PostChooser/editor-partials/modal/index.js:
```diff
@@ -25,6 +22,7 @@ export const PostChooserModal = ( props ) => {
 		postTypes,
```
```suggestion
		postTypes = [
			{ label: __( 'Posts' ), value: 'post' },
			{ label: __( 'Pages' ), value: 'page' },
		], // Default post types to search.
```

Since this `postTypes` array is used here for the selectedPostType useState(), I'd maybe define the default value here instead of in postchooser.js. Postchooser.js can just pass whatever is set in the block and still override this.

acketon commented on components/PostChooser/editor-partials/modal/index.js:
```diff
@@ -33,14 +31,17 @@ export const PostChooserModal = ( props ) => {
 	} );
 	const [ searchType, setSearchType ] = useState( 'recent' );
+	const [ selectedPostType, setSelectedPostType ] = useState(
+		postTypes && postTypes.length > 0 ? postTypes[ 0 ].value : 'post'
```
This works, but it makes an assumption that the first post type in the array is the one to show first, right? 

What about setting either a new prop like `defaultPostType = post` so the developer implementing the PostChooser would pass in the list of postTypes and which is the default post type. Or we could add `isDefault: true` property on the `postTypes` array: 

```
postTypes = [
			{ label: __( 'Posts' ), value: 'post'  },
			{ label: __( 'Pages' ), value: 'page', isDefault: true },
		], // Default post types to search.
```

acketon commented on components/PostChooser/postchooser.js:
```diff
@@ -19,7 +19,10 @@ export const PostChooser = ( props ) => {
 		modalLabel,
 		modalTitle,
-		postTypes = [ 'posts', 'pages' ], // Default post types to search.
+		postTypes = [
+			{ label: __( 'Posts' ), value: 'post' },
+			{ label: __( 'Pages' ), value: 'page' },
+		], // Default post types to search.
```
if you like the idea of moving this into the Modal component, this could just be `postTypes,` to pass in whatever is passed to this component.

acketon reviewed (approved):
@hirozed this looks great. I left a couple of optional comments for you to consider.

hirozed committed:
Updated to add the default array to the modal, added a primaryPostType

hirozed commented:
@acketon I've [made some updates](https://github.com/bu-ist/block-imports/pull/23/commits/f710896a33a9ca0243da93619682c2844385ed11)

- Moved the default `postType` array to the modal
- Added a `primaryPostType` that can be set as the default selection.

acketon reviewed (approved):
👍
